### DISCUSSION TITLE
[DONE] [vote]  - Toggle feature

### DISCUSSION THREAD

**Discussion by Olyno** [SkriptLang/Skript#4159] [Suggestions] [answered]
Hi everyone 👋🏻 

So I'm working on a new feature to be able to simply toggle some stuff (#4154). For example:
```py
set {_var} to true
toggle {_var} # {_var} is now false
```

After some discuss on Discord, some people don't find this syntax attractive, and prefer this one:

```py
set {_var} to toggled {_var}
```

So here I come: What do you guys prefer?

***The syntax with the most vote gonna be the one I will work on. Finishing Tuesday 6th.***

---
👍🏻 for the first syntax
❤️ for the second syntax
👀 for both

**Comment by APickledWalrus** [comment]
I don't really like the syntax for `set {_var} to toggled {_var}`. In this case, it would be better for it to be a word like `set {_var} to inverse of {_var}` where the expression is `inverse of {_var}` or something similar. As for the proposed TOGGLE changer, it can either be it's own thing, or an alternative for SET where it's like `toggle %booleans%` as @Mwexim essentially suggested.

**Comment by ghost** [comment]
I agree with what pickled said, i think `!` should be added too like `set {_var} to !{_var}`

**Reply by Olyno** [reply]
As Skript is suppoed to be an english language, I'm not sure that ``!`` sign is a good choice

**Reply by ghost** [reply]
> As Skript is suppoed to be an english language, I'm not sure that `!` sign is a good choice

There is `!=` many people don't even know what it means

**Comment by TPGamesNL** [comment]
One thing to keep in mind, is how this would look with expressions:
- `toggle player's flight mode` or `set player's flight mode to toggled player's flight mode`
- `toggle glowing of attacker` or `set glowing of attacker mode to toggled glowing of attacker`

**Comment by Olyno** [comment, marked as answer]
This vote is done. Thanks you to everyone who participated. The result is: both syntaxes will be added to Skript!
